scipy/optimize/tests/test_cobyqa.py

```diff
@@ -11,6 +11,7 @@
 )
 
 
+@pytest.mark.thread_unsafe
```

Comment:
Those tests are experiencing numeric mismatches:

```python
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference among violations: 5.16253706e-14
E       Max relative difference among violations: 7.74380659e-14
E        ACTUAL: array([4.955356, 0.666667])
E        DESIRED: array([4.955356, 0.666667])

```

Reply:
Okay, I can reproduce. And I think the answer to my question is: yes, `_minimize_cobyqa` needs a lock right now. This change fixes the failures:
```diff
--- a/scipy/optimize/_cobyqa_py.py
+++ b/scipy/optimize/_cobyqa_py.py
@@ -2,6 +2,11 @@ import numpy as np
 
 from ._optimize import _check_unknown_options
 
+from threading import Lock
+
+
+COBYQA_LOCK = Lock()
+
 
 def _minimize_cobyqa(fun, x0, args=(), bounds=None, constraints=(),
                      callback=None, disp=False, maxfev=None, maxiter=None,
@@ -64,4 +69,6 @@ def _minimize_cobyqa(fun, x0, args=(), bounds=None, constraints=(),
         'radius_final': float(final_tr_radius),
         'scale': bool(scale),
     }
-    return minimize(fun, x0, args, bounds, constraints, callback, options)
+
+    with COBYQA_LOCK:
+        return minimize(fun, x0, args, bounds, constraints, callback, options)
```

Can you please verify, and apply that fix and drop the `thread_unsafe` marker in the tests? This can then be followed up with opening an issue on https://github.com/cobyqa/cobyqa/ about the code not being thread-safe for free-threaded CPython. Might also happen on a default `cp313` build, since this is pure Python.

In general, we should be marking tests as `thread_unsafe` if the test code itself is the problem. If the problem is in the functionality under test, it should be made robust. And if that's too hard, adding a comment to the marker and opening an issue so that we can fix it later on.